internal/context.go

```diff
@@ -236,14 +237,7 @@ func propagateCancel(parent Context, child canceler) {
 			p.childrenLock.Unlock()
 		}
 	} else {
-		go func() {
-			s := NewSelector(parent)
-			s.AddReceive(parent.Done(), func(c ReceiveChannel, more bool) {
-				child.cancel(false, parent.Err())
-			})
-			s.AddReceive(child.Done(), func(c ReceiveChannel, more bool) {})
-			s.Select(parent)
-		}()
+		panic("parent context not found")
```

Comment:
I think this panic statement is wrong and misleading. The issue is not that the parent context is not found, the issue is the parent context is cancellable, but does not have a `*cancelCtx` in it's hierarchy. We should reflect that in the panic message

Reply:
updated the message